### ISSUE TITLE
Binary embeddings support

### ISSUE THREAD

**Issue by jimfleming** [cozodb/cozo#256]
Lots of ongoing work using binary embeddings, and other reduced precision embeddings, instead of fp32:

- https://txt.cohere.com/int8-binary-embeddings/
- https://huggingface.co/blog/embedding-quantization
- https://blog.pgvecto.rs/my-binary-vector-search-is-better-than-your-fp32-vectors

Wanted to get thoughts on the appetite for support and if there is appetite what the work might look like in case someone, maybe me, wanted to pick it up. Thanks!

**Comment by creatorrr** [comment]
yeah, support for binary embeds would be really nice. Correct me if I am wrong but adding naive support would be straightforward right? Could optimize it later on
